source/utilities.tex

```diff
@@ -19909,6 +19909,58 @@
 there is no minimum field width, and
 the field width is determined based on the content of the field.
 
+\pnum
+\indextext{string!width}%
+The \defn{width} of a string is defined as
+the estimated number of column positions appropriate
+for displaying it in a terminal.
+\begin{note}
+This is similar to the semantics of the POSIX \tcode{wcswidth} function.
+\end{note}
+
+\pnum
+For the purposes of width computation,
+a string is assumed to be in
+a locale-independent, implementation-defined encoding.
+Implementations should use a Unicode encoding
+on platforms capable of displaying Unicode text in a terminal.
+\begin{note}
+This is the case for Windows-based and many POSIX-based operating systems.
+\end{note}
+
+\pnum
+For a string in a Unicode encoding,
+implementations should estimate the width of a string
+as the sum of estimated widths of
+the first code points in its extended grapheme clusters
+as defined by
+Unicode Standard Annex \#29 Unicode Text Segmentation.
+
+\begin{example}
+The estimated width of the following code points is 2:
+\begin{codeblock}
```

Comment:
Should this be a list instead of a code region?

Reply:
I have no idea!  I don't even know if it's supposed to be an example.

Reply:
I believe this is meant to be normative text. Presentation as a list or table would work.

Reply:
Yes, this is supposed to be normative, not example.

Reply:
The list of code point ranges is part of the normative text. It is rendered as a code block in P1868 for simplicity but something like a bullet list might be a better presentation format.

Reply:
Thanks for the clarification - it's now a list.  Please have a look.